aws/pg/placement_group.go

```diff
@@ -0,0 +1,103 @@
+package pg
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/coinbase/odin/aws"
+	"github.com/coinbase/step/utils/to"
+)
+
+// FindOrCreatePartitionGroup will find a partition group by name.
+// If it doenst exist:
+// - It will create one with the provided detail (error if detail is nil)
+// If one does exist:
+// - It will fetch that Partition Group and Error if any provided detail is missing
+func FindOrCreatePartitionGroup(ec2c aws.EC2API, prefix string, groupName *string, partitionCount *int64, strategy *string) error {
+	if groupName == nil {
+		return fmt.Errorf("PlacementGroupError: groupName nil")
+	}
+
+	pg, err := findPlacementGroup(ec2c, groupName)
+	if err != nil {
+		return err
+	}
+
+	// If no pg is found create a new one
+	// otherwise validate that the found one equals the correct values
+	if pg == nil {
+		// If we are creating the Placement Group we restrict the name to prefix
+		if !strings.HasPrefix(*groupName, prefix) {
+			return fmt.Errorf("PlacementGroupError(%s): If this is a new PlacementGroup it must have %q prefix", to.Strs(groupName), prefix)
+		}
+
+		return createNewPlacementGroup(ec2c, groupName, partitionCount, strategy)
```

Comment:
Validate checks that it is available at the end of this function? Will this one be available in time?

Reply:
yes, placement groups are not much more than a name, so creation is basically instant